src/stress_test.cpp

```diff
@@ -0,0 +1,86 @@
+#include "verifier.h"
+#include <sstream>
+#include <string>
+#include <fstream>
+#include <thread>
+
+std::vector<uint8_t> HexToBytes(const char *hex_proof) {
+    int len = strlen(hex_proof);
+    assert(len % 2 == 0);
+    std::vector<uint8_t> result;
+    for (int i = 0; i < len; i += 2)
+    {
+        int hex1 = hex_proof[i] >= 'a' ? (hex_proof[i] - 'a' + 10) : (hex_proof[i] - '0');
+        int hex2 = hex_proof[i + 1] >= 'a' ? (hex_proof[i + 1] - 'a' + 10) : (hex_proof[i + 1] - '0');
+        result.push_back(hex1 * 16 + hex2);
+    }
+    return result;
+}
+
+void doit(int thread)
+{
+    std::ifstream infile("/Users/bill/downloads/newvdfs.txt");
+
+    std::string challenge;
+    std::string discriminant_size;
+    std::string input_el;
+    std::string output;
+    std::string number_of_iterations;
+    std::string witness_type;
+
+    int cnt=0;
+
+    while(cnt<20000)
+    {
+        std::getline(infile, challenge);
+        std::getline(infile, discriminant_size);
+        std::getline(infile, input_el);
+        std::getline(infile, output);
+        std::getline(infile, number_of_iterations);
+        std::getline(infile, witness_type);
+
+        if(cnt%10==thread)
+        {
+        std::vector<uint8_t> challengebytes=HexToBytes(challenge.c_str());
+        std::vector<uint8_t> inputbytes=HexToBytes(input_el.c_str());
+        std::vector<uint8_t> outputbytes=HexToBytes(output.c_str());
+
+        char *endptr;
+
+        uint64 noi=strtoll(number_of_iterations.c_str(),&endptr,10);
+        uint32 ds=strtoll(discriminant_size.c_str(),&endptr,10);
+        uint8 wt=strtoll(witness_type.c_str(),&endptr,10);
+
+        bool is_valid=CreateDiscriminantAndCheckProofOfTimeNWesolowski(challengebytes, ds, inputbytes.data(), outputbytes.data(), outputbytes.size(), noi, wt);
+
+        printf("thread %d cnt %d is valid %d %s %s %s\n",thread,cnt,is_valid,number_of_iterations.c_str(),witness_type.c_str(),challenge.c_str());
```

Comment:
you should probably terminate if `is_valid` is false. Just to make sure you read everything correctly from the file. for example, those `strtoll()` don't have any error handling.